src/main/scala/uk/gov/nationalarchives/consignmentexport/ChecksumValidator.scala

```diff
@@ -0,0 +1,32 @@
+package uk.gov.nationalarchives.consignmentexport
+
+import java.util.UUID
+
+import cats.effect.IO
+import gov.loc.repository.bagit.domain.Bag
+import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
+import uk.gov.nationalarchives.consignmentexport.Validator.ValidatedFileMetadata
+
+import scala.jdk.CollectionConverters._
+
+class ChecksumValidator(consignmentId: UUID)(implicit val logger: SelfAwareStructuredLogger[IO]) {
+
+  def validateFileChecksums(bag: Bag,
+                            validatedFileMetadata: List[ValidatedFileMetadata]): Either[RuntimeException, IO[Unit]] = {
```

Comment:
`Either[RuntimeException, IO[Unit]]` is quite an odd-looking type signature, given that `IO` can have success or failure values itself.

I think the issue is that we're mixing three different types of thing: the logic (identifying mismatched checksums), an action (logging success) and error handling (returning an exception).

I think you'd end up with code that's simpler and easier to follow if you returned the list of mismatches instead, so the return type is `List[UUID]`. Then this is a simple function, and the logging happens in the imperative part of the export code, which is in `Main`. As a bonus, it's now much easier to write a unit test of `ChecksumValidator` because it doesn't have to test for logging events or exception handling.